ui/helpers/utils/icon-factory.js

```diff
@@ -49,18 +52,19 @@ IconFactory.prototype.generateNewIdenticon = function (address, diameter) {
 
 // util
 
-function iconExistsFor(address) {
+function iconExistsFor(address, tokenList) {
   return (
-    contractMap[address] &&
+    tokenList[address] &&
     isValidHexAddress(address, { allowNonPrefixed: false }) &&
-    contractMap[address].logo
+    tokenList[address].iconUrl
   );
 }
 
-function imageElFor(address) {
-  const contract = contractMap[address];
-  const fileName = contract.logo;
-  const path = `images/contract/${fileName}`;
+function imageElFor(address, useStaticTokenList, tokenList) {
```

Comment:
Question: if `useStaticTokenList` is false, should we continue to use `contractMap`?

Reply:
Oops, I mean, if it is true?

Reply:
Oh, wait, I guess if `useStaticTokenList` is `true`, then `tokenList` will be the list from `eth-contract-metadata`?

Reply:
Because `tokenList` is managed by the controller, and we import `eth-contract-metadata` there, and make a decision about whether to use it in the controller.

Reply:
> Oh, wait, I guess if useStaticTokenList is true, then tokenList will be the list from eth-contract-metadata?

hmm... `useStaticTokenList` is set in the preferences controller, and in the token list controller... I need to think about this :thinking:

Reply:
`useStaticTokenList` is set in the preferences controller, and it is passed on to the TokenListController during its initiation along with an `onPreferencesStateChange` method.